### PR TITLE
feat: attack_counterV2 to hack set_count and add_new_owner

### PR DESCRIPTION
**Pull request by Akshola00**
# PR Description: Adding `attack_counterV2` and Required Test Functions in `test_attack_counterV2`

## Summary
This PR introduces a series of functions that hack the `counterV2` contract's `set_count` and `add_new_owner` functions.

## Files Added
- **`attack_counterV2.cairo`**: Main module that uses syscalls to hack functions of a contract.
- **`tests/test_attack_counterV2.cairo`**: Contains tests for the `attack_counterV2` contract.

## Other Changes Made
- **Tests**: Added tests for the `attack_counterV2` contract using the following test functions in `tests/test_attack_counterV2.cairo`:
  - `test_attack_counter_set_count`
  - `test_attack_counter_add_new_owner`

### CODE REVIEW

**Comment by Akshola00** [comment]
@sprtd 
synced  forked and resolved conflicts

**Comment by Akshola00** [comment]
GM @sprtd 
added test scenario where account is zero 
also added events in counter v2 contract
Ithree events: counter_set_count, counter_add_new_owner, counter_increase_count_by_one and tested them in the test_counter_v2 file and i have sucesfulltested them 
also added test for increase_count_by_one and set_count in test_counter_v2

File: src/counter_v2.cairo [hunk 1]
```diff
@@ -60,7 +60,7 @@ pub mod CounterV2 {
 
     #[derive(Copy, Drop, Debug, PartialEq, starknet::Event)]
     pub struct counter_increase_count_by_one {
```
Comment (sprtd) [inline src/counter_v2.cairo]: You should always use the [PascalCasing naming convention for structs and enums](https://book.cairo-lang.org/ch06-01-enums.html?highlight=convention#enum-variants-and-values). Rename this struct to `CountIncreased`. And also apply this change to all your `struct` and `enum` instances

Comment (Akshola00) [reply src/counter_v2.cairo]: Noted 
Will attend to these now

**Comment by Akshola00** [comment]
written test code for `student_registry` contract. pls review
@sprtd

File: tests/test_student_registry.cairo [hunk 2]
```diff
@@ -1 +1,165 @@
...
+#[should_panic(expected: 'Result::unwrap failed.')]
+fn contract_deployment_should_panic_when_deployed_with_address_zero() {
+        // deploying the student_registry contract
+        let mut student_registry_calldata: Array<felt252> = array![Accounts::zero().into()];
+        let student_registry_contract_address: ContractAddress = deploy_util("StudentRegistry", student_registry_calldata);
+        let student_registry_instance = IStudentRegistryDispatcher { contract_address: student_registry_contract_address };
+}
+
+#[test]
+fn test_add_student_successfully(){
```
Comment (sprtd) [inline tests/test_student_registry.cairo]: test for multiple successfully addition of students

Comment (Akshola00) [reply tests/test_student_registry.cairo]: done

**Pull request merged**